mesonbuild/mconf.py

```diff
@@ -302,14 +289,16 @@ def print_nondefault_buildtype_options(self):
         mismatching = self.coredata.get_nondefault_buildtype_args()
         if not mismatching:
             return
-        print("\nThe following option(s) have a different value than the build type default\n")
-        print('               current   default')
+        mlog.log("\nThe following option(s) have a different value than the build type default\n")
+        mlog.log('               current   default')
         for m in mismatching:
-            print(f'{m[0]:21}{m[1]:10}{m[2]:10}')
+            mlog.log(f'{m[0]:21}{m[1]:10}{m[2]:10}')
 
 def run(options):
     coredata.parse_cmd_line_options(options)
     builddir = os.path.abspath(os.path.realpath(options.builddir))
+    if not options.no_pager:
+        mlog.start_pager()
```

Comment:
Shouldn't a pager only be used if stdout is a tty?

Reply:
Good catch, and we already have the boilerplate for that, it's exactly what colorize_console() does.

Fixed.